controllers/capabilities/application_controller.go

```diff
@@ -268,6 +268,11 @@ func (r *ApplicationReconciler) removeApplicationFrom3scale(application *capabil
 		return nil
 	}
 
+	if account.Status.ID == nil {
```

Comment:
Will this throw a nil pointer if account.Status is nil?
```suggestion
	if account.Status != nil && account.Status.ID == nil {
```